### DISCUSSION TITLE
Replace ProtocolLib with packetevents

### DISCUSSION THREAD

**Discussion by adabugra** [QuickShop-Community/QuickShop-Hikari#1819] [Ideas]
Migrate to PacketEvents, which is faster and more regularly updated than ProtocollLib

**Comment by creatorfromhell** [comment]
> Migrate to PacketEvents, which is faster and more regularly updated than ProtocollLib

Would accept a PR for utilizing packetevents as an alternative to protcollib, but not as an entire replacement. I point to the ~47k servers running protocollib versus the ~23k using packetevents. Outside of a PR, this is something that is on the todo list, but about medium -> low priority.

**Comment by JosTheDude** [comment]
I support this myself as well, as while ProtocolLib is significantly more popular -- packetevents is MUCH better performance wise (i.e. ProtocolLib takes up 2-3% of the thread without any other plugins using it while packetevents is nearly negligble)

**Comment by creatorfromhell** [comment]
- Notes for PR:

Abstract VirtualDisplay to have a PacketProvider class, with two impls(Protocollib and PacketEvents)

**Comment by creatorfromhell** [comment]
> Migrate to PacketEvents, which is faster and more regularly updated than ProtocollLib

> I support this myself as well, as while ProtocolLib is significantly more popular -- packetevents is MUCH better performance wise (i.e. ProtocolLib takes up 2-3% of the thread without any other plugins using it while packetevents is nearly negligible)


This has been added in https://github.com/QuickShop-Community/QuickShop-Hikari/pull/1839

**Reply by JosTheDude** [reply]
this is beautiful!! thank you so much

how are version releases done and when would this be in a compiled release on modrinth?

**Reply by creatorfromhell** [reply]
> this is beautiful!! thank you so much
> 
> how are version releases done and when would this be in a compiled release on modrinth?

Won't be on modrinth for a bit, beta builds are in discord

**Reply by JosTheDude** [reply]
that was... so fast.... wtf

thank you! ill await official modrinth release for now.
